lib/kube/proxy/cluster_details.go

```diff
@@ -328,51 +340,79 @@ func getAWSResourceMatcherToCluster(kubeCluster types.KubeCluster, resourceMatch
 		if match, _, _ := services.MatchLabels(matcher.Labels, kubeCluster.GetAllLabels()); !match {
 			continue
 		}
-
-		return &(matcher.AWS)
+		return &matcher.AWS
 	}
 	return nil
 }
 
+// STSPresignClient is the subset of the STS presign interface we use in fetchers.
+type STSPresignClient = kubeutils.STSPresignClient
+
+// EKSClient is the subset of the EKS Client interface we use.
+type EKSClient interface {
+	eks.DescribeClusterAPIClient
+}
+
+// STSClient is the subset of the STS Client interface we use.
+type STSClient interface {
+	stscreds.AssumeRoleAPIClient
+}
```

Comment:
If our goal is to expose on those two interfaces, why not referring them directly? 
This is mostly internal packages and I don't see them being expanded in the future

Reply:
it is easier to have them split as one is implemented by sts.Client the other by eks.Client, used in different places.